test/e2e/scheduling/limit_range.go

```diff
@@ -41,7 +41,12 @@ const (
 var _ = SIGDescribe("LimitRange", func() {
 	f := framework.NewDefaultFramework("limitrange")
 
-	ginkgo.It("should create a LimitRange with defaults and ensure pod has those defaults applied.", func() {
+	/*
+		Release : v1.16
+		Testname: LimitRange: CRUD
+		Description: Ensure that LimitRange values are able to be applied, updated, deleted, and that this is observed by Pods.
```

Comment:
The description isn't specific enough https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-comment-metadata

Given my review of the test in question, I feel like there are too many things happening in this test to be adequately described by one `ConformanceIt`.  It's a relatively well written/scripted test though, and I would prefer to find a way to avoid adding a bunch of extra setup/teardown overhead for the various test cases.  I wonder if we can replace the `By`'s with `It`'s

Reply:
The test is fast because it uses the initial limitrange it creates for multiple assertions.

Only 7-10 seconds is spent on the test, the remaining 20-23 of that time is spent in deleting the namespace.

If we try to break them out into their own `It` / `ConformanceIt` as is, we will need to be ok with the increase/multplication in teardown time, roughly equal adding 20-30 seconds for each assertion/step we pull out into it's own test.

Another option is to look into having the asynchronous/defered teardown / namespace deletions.

Our e2e framework creates a namespace for each `It` and `Conformance.It`
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduling/limit_range.go#L42
```go
	f := framework.NewDefaultFramework("limitrange")
```

These are brought up and torn down for each test, via our outermost Before/After Each:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L132-L155

```go
// NewDefaultFramework makes a new framework and sets up a BeforeEach/AfterEach for
// you (you can write additional before/after each functions).
func NewDefaultFramework(baseName string) *Framework {
	options := Options{
		ClientQPS:   20,
		ClientBurst: 50,
	}
	return NewFramework(baseName, options, nil)
}

// NewFramework creates a test framework.
func NewFramework(baseName string, options Options, client clientset.Interface) *Framework {
	f := &Framework{
		BaseName:                 baseName,
		AddonResourceConstraints: make(map[string]ResourceConstraint),
		Options:                  options,
		ClientSet:                client,
	}

	ginkgo.BeforeEach(f.BeforeEach)
	ginkgo.AfterEach(f.AfterEach)

	return f
}
```

Digging into the AfterEach/defer a bit, we have a few options here I need to wrap my head around:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L291-L301
```go
// AfterEach deletes the namespace, after reading its events.
func (f *Framework) AfterEach() {
	RemoveCleanupAction(f.cleanupHandle)

	// DeleteNamespace at the very end in defer, to avoid any
	// expectation failures preventing deleting the namespace.
	defer func() {
		nsDeletionErrors := map[string]error{}
		// Whether to delete namespace is determined by 3 factors: delete-namespace flag, delete-namespace-on-failure flag and the test result
		// if delete-namespace set to false, namespace will always be preserved.
		// if delete-namespace is true and delete-namespace-on-failure is false, namespace will be preserved if test failed.
```